streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -247,48 +251,75 @@ public void completeRestoration() {
      */
     @Override
     public void prepareSuspend() {
-        if (state() == State.CREATED || state() == State.SUSPENDED) {
-            // do nothing
-            log.trace("Skip prepare suspending since state is {}", state());
-        } else if (state() == State.RUNNING) {
-            closeTopology(true);
+        switch (state()) {
```

Comment:
Align code style to use `switch` if all states are used

Reply:
Is this intentional to use switch and if/else in different functions?

Reply:
I think in general it's better to use `switch` and I never changed from `switch -> if/else` -- but did not update all methods either (if we only check for a single state, it seems overkill to use `switch`?

Reply:
Yup, makes sense, just curious :)